subsys/net/lib/zzhc/zzhc.c

```diff
@@ -457,6 +462,7 @@ static int check_iccid(struct zzhc *ctx)
 static int check_and_set_dns(struct zzhc *ctx)
 {
 	int err;
+	uint8_t at_resp[256];
```

Comment:
This takes a lot of stack, I suggest to put it at the top and setting it as static.

Reply:
> Please separate the latest NCS requirement fix and the DNS servers bug fix into two separate commits.

Yes, updated